### PR TITLE
Kodify test

### PR DESCRIPTION
**Pull request by danieljimenezr**
Very good and very entertaining test! My first time with doctrine ORM. My projects were with DBAL.

I have seen a txt with a rating feature, but at the readme there is no reference to this feature. So I have done the tasks at the readme (two-columns and comments).

Thanks!

### CODE REVIEW

File: src/Kodify/BlogBundle/Entity/AbstractBaseEntity.php [hunk 1]
```diff
@@ -6,80 +6,5 @@
...
-
-    /**
-     * Get updatedAt
-     *
-     * @return \DateTime
-     */
-    public function getUpdatedAt()
-    {
-        return $this->updatedAt;
-    }
```
Comment (marcostvz) [inline src/Kodify/BlogBundle/Entity/AbstractBaseEntity.php:84]: Why did you delete all those lines?

File: src/Kodify/BlogBundle/Resources/views/Post/List/index.html.twig [hunk 2]
```diff
@@ -3,10 +3,10 @@
 {% block content %}
 
     {% for post in posts %}
-        <div class="panel panel-default">
+        <div class="panel panel-default col-md-6">
             <div class="panel-heading">
                 <a href="{{ path('view_post',{id: post.id}) }}">{{ post.title }}</a>
-                <span class="pull-right">by: {{ post.author.name }} on: {{ post.createdAt|date("m/d/Y") }}</span>
```
Comment (marcostvz) [inline src/Kodify/BlogBundle/Resources/views/Post/List/index.html.twig:9]: Why did you delete the post createdAt date? :/

File: src/Kodify/BlogBundle/Resources/views/Post/List/index.html.twig [hunk 3]
```diff
@@ -3,10 +3,10 @@
 {% block content %}
 
     {% for post in posts %}
-        <div class="panel panel-default">
+        <div class="panel panel-default col-md-6">
```
Comment (marcostvz) [inline src/Kodify/BlogBundle/Resources/views/Post/List/index.html.twig:6]: Nice solution :)

If you accessed this page from a mobile device (or sth with a small screen), how do you think you'll see the posts?

File: src/Kodify/BlogBundle/Entity/Author.php [hunk 4]
```diff
@@ -112,4 +112,75 @@ public function getVideos()
...
+    {
+        // Add your code here
+    }
+
+    /**
+     * @ORM\PrePersist
+     */
+    public function initializeCreatedAt()
+    {
+        // Add your code here
```
Comment (marcostvz) [inline src/Kodify/BlogBundle/Entity/Author.php:184]: Did you forget something here? :P

File: src/Kodify/BlogBundle/Resources/views/Post/view.html.twig [hunk 5]
```diff
@@ -4,10 +4,25 @@
...
+                {% for comment in comments %}
+                    <li><b>{{ comment.text }}</b> - Commented by: <b>{{ comment.author.name }}</b></li>
+                {% endfor %}
+                </ul>
+            {% else %}
+                {{ 'There are no comments for this post. Create some!' }}
+            {% endif %}
+            <hr>
+            <div>
+                <a href="{{ path('create_post_comment', {post: post.id}) }}">Create a comment</a>
```
Comment (marcostvz) [inline src/Kodify/BlogBundle/Resources/views/Post/view.html.twig:24]: If we wanted to put the comment box here instead of a link to the create post page, how would you do it?

File: src/Kodify/BlogBundle/Entity/AbstractBaseEntity.php (hunk 1 above)
Comment (danieljimenezr) [inline src/Kodify/BlogBundle/Entity/AbstractBaseEntity.php:84]: Hi Marcos,

Well... I had some problems testing with the database and with some perms
of files (there is a commit with a comment saying it :S) and it was
appearing some errors at createAt field, so I didn't want to spend so much
time solving silly errors and I deleted this fields... I know this is a
very poor solution...

2015-12-10 13:45 GMT+01:00 Marcos Estévez Martínez <notifications@github.com

> :
> 
> In src/Kodify/BlogBundle/Entity/AbstractBaseEntity.php
> https://github.com/natxo-kodify/blog/pull/12#discussion_r47221592:
> 
> > -     \* @ORM\PrePersist
> > -     */
> > -    public function initializeCreatedAt()
> > -    {
> > -        $this->setCreatedAt(new \DateTime());
> > 
> > ## \-    }
> > -    /**
> > -     \* Get updatedAt
> > -     *
> > -     \* @return \DateTime
> > -     */
> > -    public function getUpdatedAt()
> > -    {
> > -        return $this->updatedAt;
> > -    }
> 
> Why did you delete all those lines?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/natxo-kodify/blog/pull/12/files#r47221592.

## 

Daniel
PHP & Symfony2 developer & Zitro

File: src/Kodify/BlogBundle/Resources/views/Post/List/index.html.twig (hunk 3 above)
Comment (danieljimenezr) [inline src/Kodify/BlogBundle/Resources/views/Post/List/index.html.twig:6]: Well, I hope that this bootstrap class will mantain the proportions of the div's and make a responsive reaction ;)

File: src/Kodify/BlogBundle/Entity/Author.php (hunk 4 above)
Comment (danieljimenezr) [inline src/Kodify/BlogBundle/Entity/Author.php:184]: Oh.. sorry :(

File: src/Kodify/BlogBundle/Resources/views/Post/view.html.twig (hunk 5 above)
Comment (danieljimenezr) [inline src/Kodify/BlogBundle/Resources/views/Post/view.html.twig:24]: I use to do it with ajaxRequests to the controllers. When I click at create a comment, I can render a view and attach it to the footer div with the form. 

I use to do it with absolute routing ajax or using a bundle called fsjsrouting to generate javascript routes from the routing.yml
